aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java

```diff
@@ -57,7 +57,7 @@ public class AwsClientProperties implements Serializable {
    * classes to pass provider-specific properties. Each property consists of a key name and an
    * associated value.
    */
-  private static final String CLIENT_CREDENTIAL_PROVIDER_PREFIX = "client.credentials-provider.";
+  protected static final String CLIENT_CREDENTIAL_PROVIDER_PREFIX = "client.credentials-provider.";
```

Comment:
Why change the visibility?

Reply:
Good catch: I forgot to clean the use of this property in `AwsProperties`:

```
    this.clientCredentialsProvider =
        properties.get(AwsClientProperties.CLIENT_CREDENTIALS_PROVIDER);
    this.clientCredentialsProviderProperties =
        PropertyUtil.propertiesWithPrefix(
            properties, AwsClientProperties.CLIENT_CREDENTIAL_PROVIDER_PREFIX);
```

Let me update with cleanup there.

Reply:
@Fokko @nastra if I remove `clientCredentialsProvider` from `AwsProperties`, I have to remove `AwsProperties#applyClientCredentialConfigurations()` method (to use the one defined in `AwsClientProperties`).
However, `AwsProperties#applyClientCredentialConfigurations()` is not flagged as `@Deprecated`.

I propose to just change the visibility of `CLIENT_CREDENTIAL_PROVIDER_PREFIX` in `AwsClientProperties` for now, and flag `AwsProperties#applyClientCredentialConfigurations()` with `@Deprecated` for 2.0.0 (and I will remove at that time).

Thoughts ?

Reply:
makes sense, but we should flag it to be removed in the next minor release (instead of 2.0.0)

Reply:
> I propose to just change the visibility of CLIENT_CREDENTIAL_PROVIDER_PREFIX in AwsClientProperties for now, and flag AwsProperties#applyClientCredentialConfigurations() with @Deprecated for 2.0.0 (and I will remove at that time).

That seems reasonable 👍

Reply:
Ok to flag for 1.5.0. I will do that. Thanks for your feedback !

Reply:
I updated with deprecated flag. I will do a new pass for 1.5.0 :)